apps/examples/wifi_manager/wm_test/wm_test_connectbyrssi_test.c

```diff
@@ -0,0 +1,248 @@
+		if (strncmp(wifi_scan_iter->ssid, opt->ssid, WIFIMGR_SSID_LEN) == 0) {
+			strncpy(g_bssid, wifi_scan_iter->bssid, strlen(wifi_scan_iter->bssid));
+			break;
+		}
+		wifi_scan_iter = wifi_scan_iter->next;
+	}
+	WM_TEST_SIGNAL;
+}
+
+static int wm_connect_by_rssi_init(void)
+{
+	WT_ENTER;
+	wifi_manager_result_e res = WIFI_MANAGER_SUCCESS;
+	res = wifi_manager_init(&g_wifi_callbacks);
+	if (res != WIFI_MANAGER_SUCCESS) {
+		WT_LOGE(TAG, "Failed to init wifi_manager");
+		return 1;
+	}
+	WT_LEAVE;
+	return 0;
+}
+
+static void wm_connect_by_rssi_deinit(void)
+{
+	WT_ENTER;
+	wifi_manager_result_e res = wifi_manager_deinit();
+	if (res != WIFI_MANAGER_SUCCESS) {
+		WT_LOGE(TAG, "WiFi Manager failed to stop");
+	}
+	WT_LEAVE;
+}
+
+static void _wt_scan(void *arg)
+{
+	WT_ENTER;
+	wifi_manager_result_e res = WIFI_MANAGER_SUCCESS;
+	res = wifi_manager_scan_ap(NULL);
+	if (res != WIFI_MANAGER_SUCCESS) {
+		WT_LOGE(TAG, "Failed to scan");
+		return;
+	}
+	WM_TEST_WAIT; // wait the scan result
+	WT_LEAVE;
+}
+
+static void _wt_connect(void *arg)
+{
+	WT_ENTER;
+	struct wt_options *ap_info = (struct wt_options *)arg;
+	wifi_manager_ap_config_s apconfig;
+	strncpy(apconfig.ssid, ap_info->ssid, WIFIMGR_SSID_LEN);
+	apconfig.ssid_length = strlen(ap_info->ssid);
+	apconfig.ssid[WIFIMGR_SSID_LEN] = '\0';
+	apconfig.ap_auth_type = ap_info->auth_type;
+	if (ap_info->auth_type != WIFI_MANAGER_AUTH_OPEN) {
+		strncpy(apconfig.passphrase, ap_info->password, WIFIMGR_PASSPHRASE_LEN);
+		apconfig.passphrase[WIFIMGR_PASSPHRASE_LEN] = '\0';
+		apconfig.passphrase_length = strlen(ap_info->password);
+		apconfig.ap_crypto_type = ap_info->crypto_type;
+	} else {
+		apconfig.passphrase[0] = '\0';
+		apconfig.passphrase_length = 0;
+		apconfig.ap_crypto_type = ap_info->crypto_type;
+	}
+	wt_print_wifi_ap_profile(&apconfig, "Connecting AP Info");
+	wifi_manager_result_e res = wifi_manager_connect_ap(&apconfig);
+	if (res != WIFI_MANAGER_SUCCESS) {
+		WT_LOGE(TAG, "AP connect failed");
+		return;
+	}
+	/* Wait for DHCP connection */
+	WT_LOG(TAG, "wait join done");
+	WM_TEST_WAIT;
+	WT_LEAVE;
+}
+
+static void _wt_disconnect(void *arg)
+{
+	WT_ENTER;
+	wifi_manager_result_e res = WIFI_MANAGER_SUCCESS;
+	/* Disconnect AP */
+	res = wifi_manager_disconnect_ap();
+	if (res != WIFI_MANAGER_SUCCESS) {
+		WT_LOGE(TAG, "Failed to disconnect ap (%d)", res);
+		return;
+	}
+	WM_TEST_WAIT;
+	WT_LEAVE;
+}
+
+static void wm_run_connect_by_rssi(void *arg)
+{
+	WT_ENTER;
+	int ret = -1;
+	opt = (struct wt_options *)arg;
+	ret = wm_connect_by_rssi_init();
+	if (ret) {
+		WT_LOGE(TAG, "Failed to run to check best rssi connect test ");
+		return;
+	}
+	int passed = 0, failed = 0;
+	for (int i = 0; i < opt->repeat; i++) {
+		WT_LOG(TAG, "Connection count %d", i);
+		memset(g_bssid, '\0', sizeof(g_bssid));
+		memset(g_connected_ap_mac, '\0', sizeof(g_connected_ap_mac));
+		_wt_scan(arg);
+		_wt_connect(arg);
+		if (g_bssid[0] == '\0' || g_connected_ap_mac[0] == '\0') {
```

Comment:
if g_bssid is '\0' then connect has no meaning.

Reply:
Done.